tuf/client_rework/updater_rework.py

```diff
@@ -147,71 +148,22 @@ def download_target(self, target: Dict, destination_directory: str):
         The file is saved to the 'destination_directory' argument.
         """
 
-        try:
-            for temp_obj in self._mirror_target_download(target):
+        for temp_obj in mirrors._mirror_target_download(
+            target, self._mirrors, self._fetcher
+        ):
+
+            try:
```

Comment:
commit message says it only moves code but this obviously changes things... We already have an issue on figuring out the download methods design (and fixing the bugs it has with multiple mirrors) so I'm not expecting this commit to fix everything -- but I also don't understand  what this change does?

Reply:
Thanks for catching this, it was a mistake after cherry-picking this commit from another branch.
I had to rebase and force push to fix this. This is the new commit: 35587c4